src/ascii_processor.rs

```diff
@@ -60,3 +61,57 @@ fn get_character(
         None => ch,
     }
 }
+
+///
+/// Determine how much scale to use in presence of `width` parameters,
+/// otherwise returns regular `scale` parameter per default behaviour
+///
+fn calculate_scale(args: &Arguments, dimensions: (u32, u32)) -> u32 {
```

Comment:
You can make this method as inline method using the `#[inline]` suggests, This will can provide small but easy speed win. 
https://nnethercote.github.io/perf-book/inlining.html

Reply:
No problem, even if actually is called just once per run.

Could gain more performance if we read the image from a file, or from `stdin`  (for better bash integration).

Reply:
That would mean like how the `-@` options works for `zip` program:

```bash
ls *.jpg | zip image_gallery.zip -@
```
All `*.jpg` files get streamed to `zip` and each one gets added to `image_gallery.zip`
But that would mean changing some of the program parameters.